SofaKernel/framework/sofa/core/topology/BaseMeshTopology.h

```diff
@@ -132,7 +132,9 @@ class SOFA_CORE_API BaseMeshTopology : public core::topology::Topology
     virtual const Triangle getTriangle(TriangleID i) { return getTriangles()[i]; }
     virtual const Quad getQuad(QuadID i)             { return getQuads()[i]; }
     virtual const Tetra getTetrahedron(TetraID i)    { return getTetrahedra()[i]; }
-    virtual const Hexa getHexahedron(HexaID i)       { return getHexahedra()[i]; }
+    virtual const Hexa getHexahedron(HexaID i)       { return getHexahedra()[i]; }   
+	   
+    virtual sofa::core::topology::TopologyObjectType getTopologyType() const = 0 ;
```

Comment:
Still requires a description.
For example it is unclear to me what does this function return when the topology contains both triangles and quads or tetra and hexa?
But I do no see the definition of TopologyObjectType in this PR so maybe it is something already existing and already undocumented and it is fine like that....